server/datastore/mysql/software_installers.go

```diff
@@ -148,23 +149,31 @@ func (ds *Datastore) getOrGenerateSoftwareInstallerTitleID(ctx context.Context,
 	return titleID, nil
 }
 
-func (ds *Datastore) GetSoftwareInstallerMetadata(ctx context.Context, id uint) (*fleet.SoftwareInstaller, error) {
+func (ds *Datastore) GetSoftwareInstallerMetadata(ctx context.Context, id uint, includeTitle bool) (*fleet.SoftwareInstaller, error) {
```

Comment:
maybe for the cleanup ticket but I do feel like it's okay to do a separate API call.

Reply:
Gonna go with @mna 's suggestion above and include it all the time, since otherwise I think I'd need to write a whole other method to go and fetch the title (I can't just use the existing method for getting a title because it only returns titles for software where `software_host_counts > 0`), if that works for you!

Reply:
sounds good!